provisioning_client/src/prov_device_ll_client.c

```diff
@@ -930,8 +930,8 @@ PROV_DEVICE_LL_HANDLE Prov_Device_LL_Create(const char* uri, const char* id_scop
                 else
                 {
                     // Ensure that we are passed the throttling time and send on the first send
-                    (void)tickcounter_get_current_ms(result->tick_counter, &result->status_throttle);
-                    result->status_throttle += (result->retry_after_value * 1000);
+                    (void)tickcounter_get_current_ms(result->tick_counter, &result->last_send_time);
+                    result->last_send_time += (result->retry_after_value * 1000);
```

Comment:
Nit: likely safer and less CPU to store the times in seconds.

Reply:
I agree with you. 
Not sure why this has been done like that, specially given that the code translate to seconds anyway when it uses the variable:
if ((current_time - prov_info->last_send_time) / 1000 > prov_info->retry_after_value)

And that happens on DoWork!!! (running at least every 100 milliseconds).

I'll make an exception to my focus on fixing the specific issue and change this to seconds. 

Thanks @CIPop !

Reply:
So I was about to agree with you, but...
tickcounter_get_current_ms only returns time in milliseconds.
There is no other tickcounter function for getting time in seconds.
The code needs to keep getting the time every time it calculates a time span (specially in DoWork). So it makes more sense to change the variable that holds the retry-after value to milliseconds, so we don't need to keep recalculating from milliseconds to seconds to compare.
LST, instead of going all the way with seconds, it's more advisable to go all the way with milliseconds.